src/test/java/com/example/rets_api/repository/PropertyRepositoryJPATest.java

```diff
@@ -70,9 +74,27 @@ public void should_store_a_PropertyAndSchoolAndRoom() {
         assertEquals(propertySaved.getRoomList().get(0).getRoomType(), propertyToSave.getRoomList().get(0).getRoomType());
         assertEquals(propertySaved.getRoomList().get(1).getRoomType(), propertyToSave.getRoomList().get(1).getRoomType());
 
+        assertNotNull(propertySaved.getFinancialData());
+        assertEquals(propertySaved.getFinancialData().getLeaseIndicator(), propertyToSave.getFinancialData().getLeaseIndicator());
+        assertEquals(propertySaved.getFinancialData().getLeaseOption(), propertyToSave.getFinancialData().getLeaseOption());
+        assertEquals(propertySaved.getFinancialData().getRentalAmount(), propertyToSave.getFinancialData().getRentalAmount());
+        assertEquals(propertySaved.getFinancialData().getRentalAmountCurrencyCode(), propertyToSave.getFinancialData().getRentalAmountCurrencyCode());
+        assertEquals(propertySaved.getFinancialData().getRentalAmountType(), propertyToSave.getFinancialData().getRentalAmountType());
+        assertEquals(propertySaved.getFinancialData().getTradeIndicator(), propertyToSave.getFinancialData().getTradeIndicator());
+        assertEquals(propertySaved.getFinancialData().getTradeOption(), propertyToSave.getFinancialData().getTradeOption());
+        assertEquals(propertySaved.getFinancialData().getRentalAmountPeriod(), propertyToSave.getFinancialData().getRentalAmountPeriod());
+        assertEquals(propertySaved.getFinancialData().getRentalAmountUnit(), propertyToSave.getFinancialData().getRentalAmountUnit());
+
+        assertNotNull(propertySaved.getAnimalPolicy());
+        assertEquals(propertySaved.getAnimalPolicy().getAnimalsPermitted(), propertyToSave.getAnimalPolicy().getAnimalsPermitted());
```

Comment:
you can use the method checkAllFields_AnimalPolicy from AnimalPolicyConverterTest instead replicate all code again